src/patterns/gen-ai/aws-llama-index-data-loader/index.ts

```diff
@@ -0,0 +1,493 @@
+    this.addObservabilityToConstruct(baseProps);
+
+    // Update the optional properties to their defaults
+    this.dockerImageAssetDirectory = props.dockerImageAssetDirectory ?? join(
+      __dirname, '..', '..', '..', '..', 'resources', 'gen-ai', 'aws-llama-index-data-loader', 'docker',
+    );
+    this.memoryLimitMiB = props.memoryLimitMiB ?? 2048;
+    this.containerLoggingLevel = props.containerLoggingLevel ?? 'WARNING';
+    let bucketsInvolved = [];
+    if (props.outputBucket) {
+      this.outputBucket = props.outputBucket;
+    } else {
+      this.bucketKey = new Key(this, 'LogBucketKey', {
+        enableKeyRotation: true,
+      });
+      this.logBucket = new Bucket(this, 'LogBucket', {
+        enforceSSL: true,
+        versioned: true,
+        encryption: BucketEncryption.KMS,
+        encryptionKey: this.bucketKey,
+        bucketKeyEnabled: true,
+        removalPolicy: RemovalPolicy.DESTROY,
+        autoDeleteObjects: true,
+        objectLockEnabled: true,
+        objectLockDefaultRetention: {
+          mode: ObjectLockMode.GOVERNANCE,
+          duration: Duration.days(1),
+        },
+        blockPublicAccess: BlockPublicAccess.BLOCK_ALL,
+      });
+      bucketsInvolved.push(this.logBucket);
+      this.outputBucket = new Bucket(this, 'Output', {
+        enforceSSL: true,
+        versioned: true,
+        serverAccessLogsBucket: this.logBucket,
+        serverAccessLogsPrefix: 'output-bucket-access-logs',
+        encryption: BucketEncryption.KMS,
+        encryptionKey: this.bucketKey,
+        bucketKeyEnabled: true,
+        blockPublicAccess: BlockPublicAccess.BLOCK_ALL,
+        removalPolicy: RemovalPolicy.DESTROY,
+        autoDeleteObjects: true,
+        objectLockEnabled: true,
+        objectLockDefaultRetention: {
+          mode: ObjectLockMode.GOVERNANCE,
+          duration: Duration.days(1),
+        },
+        lifecycleRules: [
+          {
+            id: 'AbortIncompleteMultipartUpload',
+            enabled: true,
+            abortIncompleteMultipartUploadAfter: Duration.days(1),
+          },
+        ],
+      });
+    }
+    bucketsInvolved.push(this.outputBucket);
+    // Create a new SSM Parameter holding a String
+    const circuitBreakerParameter = new StringParameter(this, 'CircuitBreaker', {
+      stringValue: 'False',
+    });
+    const asset = new DockerImageAsset(this, 'Image', {
+      directory: this.dockerImageAssetDirectory,
+      platform: Platform.LINUX_AMD64,
+    });
+    const queue = new Queue(this, 'Queue', {
+      visibilityTimeout: Duration.seconds(300),
+      enforceSSL: true,
+      deadLetterQueue: {
+        maxReceiveCount: 3,
+        queue: new Queue(this, 'DeadLetterQueue', {
+          enforceSSL: true,
+        }),
+      },
+    });
+    const queueProcessingFargateService = new QueueProcessingFargateService(this, 'Service', {
+      cluster: new Cluster(this, 'Cluster', {
+        containerInsights: true,
+      }),
+      vpc: props.vpc,
```

Comment:
If the vpc is passed by props or managed by the QueueProcessingFargateService, shouldn't it have a vpc endpoint for S3 ?

Reply:
The default creation, when `undefined`, does not create the S3 VPC endpoint. I can document the security practice to use VPC endpoints for the VPC

Reply:
The default creation, when `undefined`, does not create the S3 VPC endpoint. I can document the security practice to use VPC endpoints for the VPC